javascript/javascript1/week4/get-reply.js

```diff
@@ -0,0 +1,138 @@
+// voice assistant function in javascript
+/* name should be global variable beacause the name is extracted 
+in the first call to getReply and it is displayed in the second call 
+ */
+let name;
+let todo = [];
+function getReply(inputString) {
+  inputString = inputString.trim();
+  if (inputString.search("Hello my name is") == 0) {
+    // this block of code is say hello to user and store user name in the global variable
+    let positionOfName =
+      inputString.search("Hello my name is Benjamin") +
+      "Hello my name is".length;
+    name = inputString.substring(positionOfName);
+    return "Nice to meet you " + name;
```

Comment:
This is a smart way. It can also be done in other ways such as
```js
  inputString = inputString.toLowerCase();
  const commandWords = inputString.split(' ');

  if (inputString.startsWith('hello my name is')) {
    name = commandWords[commandWords.length - 1];
    return `Nice to meet you ${name}`;
  }
```